src/runtime/mod.rs

```diff
@@ -610,22 +592,23 @@ impl Module {
 
         let ctx = builder.build();
 
-        let (mut store, engine) = self.new_store_and_engine(cache_config_path, ctx)?;
+        let (mut store, engine) = self.new_store_and_engine(&info.cache_config_path, ctx)?;
         let mut linker = Linker::new(&engine);
         wasmtime_wasi::add_to_linker(&mut linker, |cx| cx)?;
 
         let http = wasi_experimental_http_wasmtime::HttpCtx::new(self.allowed_hosts.clone(), None)?;
         http.add_to_linker(&mut linker)?;
 
-        let module = self.load_cached_module(&store, cache_dir)?;
+        let module = self.load_cached_module(&store, &info.module_cache_dir)?;
         let instance = linker.instantiate(&mut store, &module)?;
 
         // Manually drop the span so we get instantiation time
         drop(startup_span);
 
+        let ep = info.entrypoint.to_owned();
         // This shouldn't error out, because we already know there is a match.
-        let start = instance.get_func(&mut store, entrypoint).ok_or_else(|| {
-            anyhow::anyhow!("No such function '{}' in {}", entrypoint, self.module)
+        let start = instance.get_func(&mut store, info.entrypoint.as_str()).ok_or_else(|| {
```

Comment:
Nit: You can just do `&info.entrypoint` to pass a `String` as an `&str`

Reply:
done